src/Processors/QueryPlan/Optimizations/optimizeReadInOrder.cpp

```diff
@@ -504,10 +491,14 @@ AggregationInputOrder buildInputOrderInfo(
     using ReverseMatches = std::unordered_map<const ActionsDAG::Node *, MatchedTrees::Matches::const_iterator>;
     ReverseMatches reverse_matches;
 
+    /// Map from key_name to its positions in key_names.
+    std::unordered_map<std::string_view, Positions> not_matched_keys;
```

Comment:
what is the case when we'd need several positions per key? (i.e. why not size_t instead of Positions)

Reply:
Added comment:

https://github.com/ClickHouse/ClickHouse/blob/1f774ee1a5541dc989b6a11e120002198d7dc243/src/Processors/QueryPlan/Optimizations/optimizeReadInOrder.cpp#L491-L494